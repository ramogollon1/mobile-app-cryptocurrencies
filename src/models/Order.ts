import { ReactNode } from 'react'

export enum OperationType {
  BUY = 'BUY',
  SELL = 'SELL',
}

export enum Cryptocurrency {
  BTC = 'BTC',
  ETH = 'ETH',
  USDC = 'USDC',
}

export enum FiatCurrency {
  USD = 'USD',
  USDT = 'USDT',
}
export enum OrderMode {
  LIMIT = 'limit',
  MARKET = 'market',
}

export enum OrderStatus {
  EXECUTED = 'Executed',
  PENDING = 'Pending',
}

export interface CurrencyRate {
  symbol: string
  price: string
}

export type Order = {
  id: number
  operationType: OperationType
  cryptocurrency: Cryptocurrency
  price: number
  amount: number
  orderType: OrderMode
  isExecuted: boolean
  commission?: number
  total?: number
}

export type ExtendedOrder = Order & {
  total: number
  commission: number
}

export interface OrderContextData {
  orders: Order[]
  addOrder: (order: Order) => void
  executeOrder: (orderId: number) => void
  totalCommission: number
}

export interface OrderProviderProps {
  children: ReactNode
}
