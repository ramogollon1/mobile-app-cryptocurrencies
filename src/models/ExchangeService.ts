import { CurrencyRate } from './Order'

export interface ExchangeService {
  getCurrencyRates(): Promise<CurrencyRate[]>
}
