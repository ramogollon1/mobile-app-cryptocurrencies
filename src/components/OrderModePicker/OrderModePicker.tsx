import React from 'react'
import { Picker } from '@react-native-picker/picker'
import { OrderMode } from '../../models/Order'
import { ViewStyle } from 'react-native'

interface OrderModePickerProps {
  orderType: OrderMode
  setOrderType: (orderType: OrderMode) => void
  style?: ViewStyle
}

const orderModes: OrderMode[] = [OrderMode.LIMIT, OrderMode.MARKET]

export const OrderModePicker: React.FC<OrderModePickerProps> = ({
  orderType,
  setOrderType,
  style,
}) => (
  <Picker
    style={style}
    selectedValue={orderType}
    onValueChange={itemValue => setOrderType(itemValue)}>
    {orderModes.map(mode => (
      <Picker.Item key={mode} label={mode} value={mode} testID={mode} />
    ))}
  </Picker>
)
