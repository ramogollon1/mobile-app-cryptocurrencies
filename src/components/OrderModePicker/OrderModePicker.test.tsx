import React from 'react'
import renderer from 'react-test-renderer'
import { OrderModePicker } from './OrderModePicker'
import { OrderMode } from '../../models/Order'

describe('OrderModePicker', () => {
  it('renders correctly', () => {
    const setOrderType = jest.fn()

    const tree = renderer
      .create(
        <OrderModePicker
          orderType={OrderMode.LIMIT}
          setOrderType={setOrderType}
        />,
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
