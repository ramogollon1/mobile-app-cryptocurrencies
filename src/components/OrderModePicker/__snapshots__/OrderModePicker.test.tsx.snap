// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderModePicker renders correctly 1`] = `
<View>
  <RNCPicker
    items={
      [
        {
          "label": "limit",
          "testID": "limit",
          "textColor": undefined,
          "value": "limit",
        },
        {
          "label": "market",
          "testID": "market",
          "textColor": undefined,
          "value": "market",
        },
      ]
    }
    numberOfLines={1}
    onChange={[Function]}
    selectedIndex={0}
    style={
      [
        {
          "height": 216,
        },
        undefined,
      ]
    }
  />
</View>
`;
